src/plugins/discover/public/components/discover_grid/discover_grid.tsx

```diff
@@ -446,8 +451,11 @@ export const DiscoverGrid = ({
     return { columns: sortingColumns, onSort: () => {} };
   }, [sortingColumns, onTableSort, isSortEnabled]);
   const lead = useMemo(
-    () => getLeadControlColumns(setExpandedDoc).filter(({ id }) => controlColumnIds.includes(id)),
-    [controlColumnIds, setExpandedDoc]
+    () =>
+      getLeadControlColumns(!!(setExpandedDoc && DocumentView)).filter(({ id }) =>
```

Comment:
Nit: This we should add a variable named like you named the param of `getLeadControlColumns` before L453 like this

`const canSetExpandedDoc = Boolean(setExpandedDoc && DocumentView);`

and also use it in L636